doc/system-telemetry/reboot-cause.md

```diff
@@ -0,0 +1,118 @@
+# Reboot-cause information via telemetry agent
+
+## Revision
+
+| Rev | Date     | Author      | Change Description |
+|:---:|:--------:|:-----------:|--------------------|
+| 0.1 | 09/02/20 | Sujin Kang  | Initial version    |
+
+## Scope
+-Enable sonic streaming telemetry agent to send Reboot-cause information
+
+### Enable sonic streaming telemetry agent to send Reboot-cause information
+
+##### Part 1 
+For 1st part, process-reboot-cause copies the previous-reboot-cause.txt 
+to "/host/reboot-cause/previous-reboot-cause/" with adding timestamp 
+at the end of file name after processing the reboot-cause.
+Currently previous-reboot-cause.txt is plain text format but this file content will be formatted to be parsed easily.
+Read each reboot-cause information from saved previous-reboot-cause files
+And update reboot-cause information up to 10 entries to state-DB.  
+
+The example shows the previous reboot-cause files stored in /host/reboot-cause/previous-reboot-cause/.
+```
+$ls /host/reboot-cause/previous-reboot-cause/
+previous-reboot-cause-20200903T232033.txt
+previous-reboot-cause-20200902T101105.txt
+previous-reboot-cause-20200902T015048.txt
+```
+The following example shows the content of the previous reboot-cause file - previous-reboot-cause-20200903T232033.txt.
+```
+TIMESTAMP: "20200903T232033"
+CAUSE: "reboot"
+USER: "admin"
+TIME: "Thu 03 Sep 2020 11:15:30 PM UTC"
+COMMENT: "User issued 'reboot' command [User: admin, Time: Thu 03 Sep 2020 11:15:30 PM UTC]" 
+```
+
+Details of CLI and state-DB given below. 
+Cli command to retrieve the Reboot-cause information
+```
+$ show reboot-cause
+```
+state-DB schema to store the Reboot-cause information
+```
+; Defines information for reboot-cause
+key                     = REBOOT_CAUSE|timestamp         ; last reboot-cause processing time
```

Comment:
Suggest adding angle brackets to indicate timestamp is variable: `REBOOT_CAUSE|<timestamp>`